packages/modelviewer.dev/examples/scenegraph/index.html

```diff
@@ -560,6 +560,111 @@ <h2 class="demo-title">Transform textures</h2>
     </div>
 
 
+    <div class="sample">
+      <div id="hideShowMeshVariantsExample" class="demo"></div>
+      <div class="content">
+        <div class="wrapper">
+          <div class="heading">
+            <h2 class="demo-title">Toggle between Mesh Variants</h2>
+            <p>If you have a few mesh variants, you can hide and show them by changing textures opacity.</p>
+          </div>
+          <example-snippet stamp-to="hideShowMeshVariantsExample" highlight-as="html">
+            <template>
+<model-viewer id="manifold" camera-controls touch-action="pan-y" src="../../shared-assets/models/manifold.glb" ar alt="A 3D model of a stick and ball">
+  <div class="controls">
+    <div>Mesh Variants: <select id="textures"></select>
+    </div>
+</model-viewer>
+<script type="module">
+const mvTextures = document.querySelector("model-viewer#manifold");
+const textures = document.querySelector('#textures');
+
+mvTextures.addEventListener("load", () => {
+  const materials = mvTextures.model.materials
+  for(const material of materials) {
+    const option = document.createElement('option')
+    option.value = material.name;
+    option.textContent = material.name;
+    textures.appendChild(option);
+  }
+
+  let startTimestamp;
+  const animationDuration = 1000; // 1 second
+  let alpha0 = 1;
+  let alpha1 = 0;
+
+  function updateAlphaValues(texture0, texture1, elapsed) {
+      alpha0 = 1 - (elapsed / animationDuration);
+      alpha1 = elapsed / animationDuration;
+      texture0.pbrMetallicRoughness.setBaseColorFactor([
+        texture0.pbrMetallicRoughness.baseColorFactor[0],
+        texture0.pbrMetallicRoughness.baseColorFactor[1],
+        texture0.pbrMetallicRoughness.baseColorFactor[2],
+        alpha0,
+      ]);
+      texture1.pbrMetallicRoughness.setBaseColorFactor([
+        texture1.pbrMetallicRoughness.baseColorFactor[0],
+        texture1.pbrMetallicRoughness.baseColorFactor[1],
+        texture1.pbrMetallicRoughness.baseColorFactor[2],
+        alpha1,
+      ]);
+    }
+
+    mvTextures.addEventListener("load", () => {
+      const materials = mvTextures.model.materials;
+      for (const material of materials) {
+        const option = document.createElement('option');
+        option.value = material.name;
+        option.textContent = material.name;
+        textures.appendChild(option);
+      }
+    });
+
+    textures.addEventListener('input', (event) => {
+      const selectedValue = event.target.value;
+      const ball = mvTextures.model.materials[0];
```

Comment:
Let's use `model.getMaterialByName` instead - if you put the names in your HTML options you can even iterate over those.

Reply:
I used `model.getMaterialByName` lmk if you prefer the other approach.